packages/hub/tasks/notify-merchant-claim.ts

```diff
@@ -0,0 +1,42 @@
+import { inject } from '@cardstack/di';
+import CardpaySDKService from '../services/cardpay-sdk';
+import Web3HttpService from '../services/web3-http';
+import WorkerClient from '../services/worker-client';
+
+interface MerchantClaimSubscriptionEvent {
+  returnValues: {
+    merchantSafe: string;
+    amount: string;
+  };
+}
+
+export default class NotifyMerchantClaim {
+  cardpay: CardpaySDKService = inject('cardpay');
+  web3: Web3HttpService = inject('web3-http', { as: 'web3' });
+  workerClient: WorkerClient = inject('worker-client', { as: 'workerClient' });
+
+  async perform(payload: MerchantClaimSubscriptionEvent) {
+    let merchantSafeAddress = payload.returnValues.merchantSafe;
+    let amountInWei = payload.returnValues.amount;
+
+    let web3Instance = this.web3.getInstance();
+    let Safes = await this.cardpay.getSDK('Safes', web3Instance);
+    let { safe } = await Safes.viewSafe(merchantSafeAddress);
+
+    if (!safe) {
+      // Sufficient to log to Sentry because of on('job:error')…?
+      throw new Error('Safe not found');
+    }
+
+    // TODO: how should we resolve the token symbol from an address?
+    // let tokenAddress = event.returnValues.payableToken;
+    let token = 'DAI.CPXD';
```

Comment:
the subgraph has this information

Reply:
Here is an example illustrating how you might call this using a txn hash (which should be available from the event)
![Screen Shot 2021-11-30 at 4 17 33 PM](https://user-images.githubusercontent.com/61075/144328880-4b9bf326-206e-4529-81eb-cd0409601b4d.png)

Reply:
The SDK has an API for issuing direct graphQL queries